File: demo-liquid-glass/DemoLiquidGlass/domain/models/Assessment.swift
// domain/models/Assessment.swift
// Pure Swift struct — NO SwiftUI, NO Foundation imports

public struct Assessment: Identifiable, Equatable, Hashable {
    public let id: String
    public let title: String
    public let courseId: String
    public let deadline: String       // ISO-8601 string — avoids Foundation.Date in domain
    public let criteriaIds: [String]
    public let peerIds: [String]

    public init(
        id: String,
        title: String,
        courseId: String,
        deadline: String,
        criteriaIds: [String],
        peerIds: [String]
    ) {
        self.id = id
        self.title = title
        self.courseId = courseId
        self.deadline = deadline
        self.criteriaIds = criteriaIds
        self.peerIds = peerIds
    }
}

File: demo-material3/lib/features/peer_evaluation/domain/models/assessment.dart
// Pure Dart — NO Flutter imports
class Assessment {
  final String id;
  final String courseId;
  final String title;
  final DateTime deadline;
  final List<String> criteriaIds;
  final List<String> peerIds;

  const Assessment({
    required this.id,
    required this.courseId,
    required this.title,
    required this.deadline,
    required this.criteriaIds,
    required this.peerIds,
  });
}
